exp/controllers/awsmachinepool_controller.go

```diff
@@ -214,14 +215,14 @@ func (r *AWSMachinePoolReconciler) reconcileNormal(ctx context.Context, machineP
 	if !machinePoolScope.Cluster.Status.InfrastructureReady {
 		machinePoolScope.Info("Cluster infrastructure is not ready yet")
 		conditions.MarkFalse(machinePoolScope.AWSMachinePool, expinfrav1.ASGReadyCondition, infrav1.WaitingForClusterInfrastructureReason, clusterv1.ConditionSeverityInfo, "")
-		return ctrl.Result{}, nil
+		return ctrl.Result{Requeue: true}, nil
```

Comment:
why is this returning true now? that would active the rate limiter, wouldn't this be just requeue when a change in the status triggers and event?

Reply:
Done it based on discussion here: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4321#issuecomment-1588846160